lib/rucio/daemons/reaper/reaper2.py

```diff
@@ -177,7 +180,10 @@ def delete_from_storage(replicas, prot, rse_info, staging_areas, auto_exclude_th
                     # sign the URL if necessary
                     if prot.attributes['scheme'] == 'https' and rse_info['sign_url'] is not None:
                         pfn = get_signed_url(rse_id, rse_info['sign_url'], 'delete', pfn)
-                    prot.delete(pfn)
+                    if not bulk:
+                        prot.delete(pfn)
+                    else:
+                        replicas_to_bulk_delete.append(replica)
```

Comment:
I don't think we must act on replicas here. There seem to be an existing separation between the core (working on replicas), and protocols (acting on pfns). You only need pfns anyway inside the protocol class.